include/zephyr/rtio/rtio.h

```diff
@@ -1280,17 +1351,25 @@ static inline int z_impl_rtio_cqe_copy_out(struct rtio *r,
 					   size_t cqe_count,
 					   k_timeout_t timeout)
 {
-	size_t copied;
+	size_t copied = 0;
 	struct rtio_cqe *cqe;
+	uint64_t end = sys_clock_timeout_end_calc(timeout);
 
-	for (copied = 0; copied < cqe_count; copied++) {
-		cqe = rtio_cqe_consume_block(r);
+	do {
+		cqe = K_TIMEOUT_EQ(timeout, K_FOREVER) ? rtio_cqe_consume_block(r)
+						       : rtio_cqe_consume(r);
 		if (cqe == NULL) {
-			break;
+#ifdef CONFIG_BOARD_NATIVE_POSIX
```

Comment:
Only the Z_SPIN_DELAY(1) should be needed, it does the right thing if posix is used I believe? Or is the problem the clock doesn't move?

Reply:
Right, the issue is the clock doesn't move.